Entity/TaxRateInterface.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Commerce\BasketBundle\Entity;
+
+/**
+ * Interface TaxRateInterface implements minimal needed functions for tax rate classes.
+ */
+interface TaxRateInterface
+{
+    public static function getInstance();
+    public function getValue();
```

Comment:
```suggestion
   public static function getInstance(): TaxRateInterface;
    public function getValue(): float;
```
Wenn du ein Interface für die TaxRates definierst solltest du explizit die Rückgabewerte definieren. In deinem Beispiel könnte eine TaxRate das Interface implementieren aber z.b. ein String bei dem `getValue()` Aufruf zurückgeben, wodurch die Preisberechnung brechen würde.